x-pack/plugins/triggers_actions_ui/public/application/sections/rule_form/rule_form_advanced_options.tsx

```diff
@@ -0,0 +1,316 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback, useMemo } from 'react';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+import {
+  EuiBadge,
+  EuiFieldNumber,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFormRow,
+  EuiIconTip,
+  EuiPanel,
+  EuiSwitch,
+  EuiText,
+  useIsWithinMinBreakpoint,
+  useEuiTheme,
+  EuiHorizontalRule,
+  EuiSpacer,
+  EuiSplitPanel,
+  EuiLoadingSpinner,
+} from '@elastic/eui';
+import { RuleSettingsFlappingInputs } from '@kbn/alerts-ui-shared/src/rule_settings/rule_settings_flapping_inputs';
+import {
+  RuleSettingsFlappingMessage,
+  flappingOffMessage,
+} from '@kbn/alerts-ui-shared/src/rule_settings/rule_settings_flapping_message';
+import { RuleSpecificFlappingProperties } from '@kbn/alerting-plugin/common';
+import { useGetFlappingSettings } from '../../hooks/use_get_flapping_settings';
+
+const flappingLabel = i18n.translate(
+  'xpack.triggersActionsUI.ruleFormAdvancedOptions.flappingLabel',
+  {
+    defaultMessage: 'Flapping Detection',
+  }
+);
+
+const flappingOnLabel = i18n.translate('xpack.triggersActionsUI.ruleFormAdvancedOptions.onLabel', {
+  defaultMessage: 'ON',
+});
+
+const flappingOffLabel = i18n.translate(
+  'xpack.triggersActionsUI.ruleFormAdvancedOptions.offLabel',
+  {
+    defaultMessage: 'OFF',
+  }
+);
+
+const flappingOverrideLabel = i18n.translate(
+  'xpack.triggersActionsUI.ruleFormAdvancedOptions.overrideLabel',
+  {
+    defaultMessage: 'Override',
+  }
+);
+
+const flappingOverrideConfiguration = i18n.translate(
+  'xpack.triggersActionsUI.ruleFormAdvancedOptions.flappingOverrideConfiguration',
+  {
+    defaultMessage: 'Override Configuration',
+  }
+);
+
+const clampFlappingValues = (flapping: RuleSpecificFlappingProperties) => {
+  return {
+    ...flapping,
+    statusChangeThreshold: Math.min(flapping.lookBackWindow, flapping.statusChangeThreshold),
+  };
+};
+
+const INTEGER_REGEX = /^[1-9][0-9]*$/;
+
+export interface RuleFormAdvancedOptionsProps {
+  alertDelay?: number;
+  flappingSettings?: RuleSpecificFlappingProperties;
+  onAlertDelayChange: (value: string) => void;
+  onFlappingChange: (value: RuleSpecificFlappingProperties | null) => void;
+  enabledFlapping?: boolean;
+}
+
+export const RuleFormAdvancedOptions = (props: RuleFormAdvancedOptionsProps) => {
+  const {
+    alertDelay,
+    flappingSettings,
+    enabledFlapping = false,
+    onAlertDelayChange,
+    onFlappingChange,
+  } = props;
+
+  const isDesktop = useIsWithinMinBreakpoint('xl');
+
+  const { euiTheme } = useEuiTheme();
+
+  const { data: spaceFlappingSettings, isInitialLoading } = useGetFlappingSettings({
```

Comment:
What does `spaceFlappingSettings` mean? How does it differ from the `flappingSettings` in the props? Also, what's the type? 😅

Reply:
space flapping setting is the "global" flapping setting. The `flappingSettings` then becomes the rule specific flapping setting.

They both should be the same type